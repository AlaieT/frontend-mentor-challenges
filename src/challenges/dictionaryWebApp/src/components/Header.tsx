import React from "react";

import FontSelect from "./FontSelect";
import ModeChange from "./ModeChange";

import { ReactComponent as LogoIcon } from "../assets/images/logo.svg";

import type { HeaderProps } from "../types";

import styles from "../styles/components/header.module.scss";

const Header = ({
  font,
  mode,
  handleFont = () => undefined,
  handleMode = () => undefined
}: HeaderProps) => (
  <header id={styles.header}>
    <LogoIcon width={32} height={36.5} />
    <div id={styles.controls}>
      <FontSelect callback={handleFont} font={font} mode={mode} />
      <span id={styles.separate} />
      <ModeChange callback={handleMode} mode={mode} />
    </div>
  </header>
);

export default Header;
