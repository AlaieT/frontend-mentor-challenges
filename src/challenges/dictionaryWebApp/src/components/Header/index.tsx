import React from "react";

import { GlobalContext } from "../../utils/context";

import { ReactComponent as LogoIcon } from "../../assets/images/logo.svg";
import { ReactComponent as MoonIcon } from "../../assets/images/icon-moon.svg";

import styles from "../../styles/components/header.module.scss";

const Header = () => {
  const globalContext = React.useContext(GlobalContext);

  return (
    <header>
      <LogoIcon width={32} height={36.5} />
      <div id={styles.controls}>
        <div id={styles.mode}>
          <MoonIcon width={20} height={20} />
        </div>
      </div>
    </header>
  );
};

export default Header;
