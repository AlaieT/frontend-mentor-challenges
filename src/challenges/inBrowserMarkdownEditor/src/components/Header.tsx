import React, { useRef, useState } from "react";

import { ReactComponent as IconMenu } from "../assets/images/icon-menu.svg";
import { ReactComponent as IconClose } from "../assets/images/icon-close.svg";
import { ReactComponent as IconDelete } from "../assets/images/icon-delete.svg";
import { ReactComponent as IconSave } from "../assets/images/icon-save.svg";
import { ReactComponent as IconDocument } from "../assets/images/icon-document.svg";
import { ReactComponent as Logo } from "../assets/images/logo.svg";
import styles from "../styles/components/header.module.scss";
import { HeaderProps } from "../types";

const Header = ({
  isMenuOpen,
  handleOpenMenu,
  handleDocumentName,
  documentName = "welcome.md",
  handleDelete,
  handleSave
}: HeaderProps) => {
  const nameInputRef = useRef<HTMLInputElement>(null);
  const [isNameChange, setIsNameChange] = useState(false);
  const handleIsNameChange = (
    e:
      | React.MouseEvent<HTMLButtonElement>
      | React.FocusEvent<HTMLInputElement, Element>
  ) => {
    e.preventDefault();
    setIsNameChange((item) => !item);

    if (isNameChange) {
      handleDocumentName(nameInputRef.current?.value || "");
    }
  };

  return (
    <header id={styles.header}>
      <button id={styles["menu-button"]}>
        {isMenuOpen ? <IconClose /> : <IconMenu />}
      </button>
      <div id={styles.title}>
        <Logo />
      </div>
      <span id={styles.split} />
      <div id={styles["document-name"]}>
        <IconDocument/>
        <div id={styles["name-input"]}>
          <p>Document Name</p>
          <div id={styles["input-button-area"]}>
            {isNameChange ? (
              <input
                ref={nameInputRef}
                // eslint-disable-next-line jsx-a11y/no-autofocus
                autoFocus
                defaultValue={documentName}
                onBlur={handleIsNameChange}
              />
            ) : (
              <button onClick={handleIsNameChange}>{documentName}</button>
            )}
          </div>
        </div>
      </div>
      <button id={styles.delete}>
        <IconDelete />
      </button>
      <button id={styles.save}>
        <IconSave width={16} height={16} />
        <p>Save Changes</p>
      </button>
    </header>
  );
};

export default Header;
