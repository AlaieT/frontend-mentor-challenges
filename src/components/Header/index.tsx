import React from "react";

import { ReactComponent as GitHubIcon } from "../../assets/images/github.svg";
import { ReactComponent as ArrowIcon } from "../../assets/images/arrow_outward.svg";
import styles from "../../styles/components/header.module.scss";

const Header = () => (
  <header id={styles.header}>
    <div id={styles.author}>
      <p>ilya [alaie] shafeev</p>
      <p>Â©2023</p>
    </div>
    <div id={styles.greetings}>
      <div id={styles.title}>
        <p>front-end mentor</p>
        <p>challenges</p>
      </div>
      <div id={styles.description}>
        <p>
          This website contains my solutions for Frontend Mentor challenges. All
          challenges made as individual web projects, so they include using of
          frameworks, linters, testing and e.t.c. to make practice building
          websites in a realistic workflow. All of these challenges help improve
          skills and gain experience creating websites.
        </p>
      </div>
      <a id={styles.github} href="/" target="_blank" rel="noopener noreferrer">
        <GitHubIcon width={20} height={20} />
        GITHUB
        <ArrowIcon width={20} height={20} />
      </a>
    </div>
  </header>
);

export default Header;
