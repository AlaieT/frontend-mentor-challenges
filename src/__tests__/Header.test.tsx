import React from "react";
import { render, cleanup, act } from "@testing-library/react";

import Header from "../components/Header";

afterEach(cleanup)

describe("Header", () => {
  const titleProps = {
    sentence: "frontend mentorâ€”challenges",
    delay: 50
  };

  describe("correct render", () => {
    it("should match snapshot before animation ends", () => {
      expect(render(<Header challengesCount={15}/>).asFragment()).toMatchSnapshot();
    });

    it("should match snapshot after animation ends", async () => {
      const { asFragment } = render(<Header challengesCount={15}/>);

      await act(async () => {
        await new Promise((resolve, _) =>
          setTimeout(
            () => resolve("done"),
            titleProps.sentence.length * titleProps.delay + 50
          )
        );
      });

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
