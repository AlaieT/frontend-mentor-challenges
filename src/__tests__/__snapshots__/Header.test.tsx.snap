// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correct render should match snapshot 1`] = `
<DocumentFragment>
  <header
    id="header"
  >
    <div
      id="author"
    >
      <p>
        ilya [alaie] shafeev
      </p>
      <p>
        Â©2023
      </p>
    </div>
    <div
      id="greetings"
    >
      <div
        id="title"
      >
        <p>
          front-end mentor
        </p>
        <p>
          challenges
        </p>
      </div>
      <div
        id="description"
      >
        <p>
          This website contains my solutions for Frontend Mentor challenges. All challenges made as individual web projects, so they include using of frameworks, linters, testing and e.t.c. to make practice building websites in a realistic workflow. All of these challenges help improve skills and gain experience creating websites.
        </p>
      </div>
      <a
        href="/"
        id="github"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          height="20"
          width="20"
        />
        GITHUB
        <div
          height="20"
          width="20"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
